LibreHardwareMonitor/Utilities/Logger.cs

```diff
@@ -170,13 +181,30 @@ public void Log()
         if (_lastLoggedTime + LoggingInterval - new TimeSpan(5000000) > now)
             return;
 
-        if (_day != now.Date || !File.Exists(_fileName))
+        switch (FileRotationMethod)
         {
-            _day = now.Date;
-            _fileName = GetFileName(_day);
-
-            if (!OpenExistingLogFile())
-                CreateNewLogFile();
+            case LoggerFileRotationMethod.PerSession:
+                // Create file if it does not exist or the logging interval has passed (+ some margin)
+                if (!File.Exists(_fileName) || now - _lastLoggedTime > (LoggingInterval + TimeSpan.FromMilliseconds(100)))
+                {
+                    uint sessionNumber = 0;
```

Comment:
This logic is not right, you're now also having the first session the same as the day one. The first session number should start at 1.

Reply:
Good catch